scan/lib/scan/test_command_generator.rb

```diff
@@ -48,9 +48,20 @@ def actions
 
         actions = []
         actions << :clean if config[:clean]
-        actions << :build unless config[:skip_build]
-        actions << :test
 
+        building_mode = :test
+        if config[:build_for_testing]
+          building_mode = "build-for-testing"
+          config[:skip_build] = true
+        end
+
+        if config[:test_without_building]
+          building_mode = "test-without-building"
+          config[:skip_build] = true
+        end
+
+        actions << :build unless config[:skip_build]
+        actions << building_mode
```

Comment:
Hey @hjanuschka, I think ideally we would be able to avoid manipulating the `config` object. Would it be okay to re-write to be a bit more like so:

```ruby
if config[:build_for_testing]
  actions << "build-for-testing"
elsif config[:test_without_building]
  actions << "test-without-building"
else
  actions << :build unless config[:skip_build]
  actions << :test
end
```